data_processors/pipeline/lambdas/dragen_wts.py

```diff
@@ -121,6 +122,16 @@ def handler(event, context) -> dict:
         subject_id=subject_id)
     workflow_engine_parameters = wfl_helper.get_engine_parameters(target_id=subject_id, secondary_target_id=None)
 
+    if event.get('arriba_large_mem', False):
+        arriba_overrides = ICAResourceOverridesStep("#arriba_fusion_step", "standardHiMem", "medium")
+        if not "overrides" in workflow_engine_parameters.keys():
+            workflow_engine_parameters["overrides"] = {}
+        if not arriba_overrides.step_id in workflow_engine_parameters["overrides"].keys():
+            workflow_engine_parameters["overrides"][arriba_overrides.step_id] = {}
+        if not "requirements" in workflow_engine_parameters["overrides"][arriba_overrides.step_id].keys():
+            workflow_engine_parameters["overrides"][arriba_overrides.step_id]["requirements"] = {}
+        workflow_engine_parameters["overrides"][arriba_overrides.step_id]["requirements"].update(arriba_overrides.get_resource_requirement_overrides())
+
```

Comment:
May I suggest, 
* We split this if down under code block into its own function somewhere down below. 
* And then, to add unit test to that function for each "if" cases, please. 

Probably, need to mock `SecondaryAnalysisHelper` object. The corresponding test file is in `tests/test_dragen_wts.py`.